File: 프로그래머스/lv0/120863. 다항식 더하기/다항식 더하기.js
function solution(polynomial) {
    const variable = polynomial.split(' + ');
    
    let xnum = 0;
    let n = 0;
    
    variable.forEach(el => {
        if(el.includes('x')){
            xnum += Number(el.slice(0, -1) || 1)
        }else n += Number(el);
    })
    
    if(xnum === 0) return `${n}`;
    else if(xnum === 1) return n === 0 ? 'x' : `x + ${n}`;
    else return n === 0 ? `${xnum}x` : `${xnum}x + ${n}`;
}

File: 프로그래머스/lv0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    splited = polynomial.split(' + ');
    n = sum([int(i) for i in splited if 'x' not in i])
    xnum = sum([1 if i == 'x' else int(i[0:-1]) for i in splited if 'x' in i])
    
    if xnum == 0:
        return str(n)
    elif xnum == 1:
        return 'x + '+str(n) if n != 0 else 'x'
    else:
        return str(xnum)+'x + '+ str(n) if n != 0 else str(xnum)+'x'